app/models/spree/order_decorator.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+Spree::Order.class_eval do
+  def available_payment_methods
+    @available_payment_methods ||= Spree::PaymentMethod
+        .active
+        .available_to_store(store)
+        .available_to_users
+        .left_joins(:payment_method_zone)
+        .where(spree_payment_method_zones: {zone_id: [nil] + Spree::Zone.for_address(ship_address).map(&:id)})
```

Comment:
We could use `Spree::Zone.for_address(ship_address).pluck(:id)` here instead of mapping